WeatherMonitoringSystem/Domain/Core/WeatherStation.cs

```diff
@@ -0,0 +1,43 @@
+﻿using WeatherMonitoringSystem.Bots;
+
+namespace WeatherMonitoringSystem.Core
+{
+    public class WeatherStation
+    {
+        private List<IWeatherBot> _bots = new();
+        private WeatherData _currentWeather;
+
+        public void AddBot(IWeatherBot bot)
+        {
+            _bots.Add(bot);
+        }
+
+        public void RemoveBot(IWeatherBot bot)
+        {
+            _bots.Remove(bot);
+        }
+
+        public List<IWeatherBot> SetWeatherData(WeatherData data)
+        {
+            _currentWeather = data;
+            return NotifyBots();
+        }
+
+        private List<IWeatherBot> NotifyBots()
```

Comment:
The name of this method doesn't indicate that it will only return the activated bots, it's a bit misleading, you can instead just return all bots here, or a separate method to return the bots.